wordpress-mobile/WordPress-Android
# Add Support for UI Testing the Jetpack app and Compose layouts

This PR:
- Adds support for testing Compose UI
- Adapts the e2e tests to support running on the Jetpack app
- Adds support for running the e2e tests on the upcoming redesigned Landing screen of the WordPress app, while still supporting the current landing screen.

## To test:
- ✅ **Verify** the instrumented tests passed CI
- 1️⃣ Optionally to validate the e2e tests support Jetpack app, you can run the suite locally in Android Studio: 
  1. Change active build variant to: `jetpackWasabiDebug`
  2. Run all tests in `WordPress/src/androidTest/java/org/wordpress/android/e2e`
      > **Note** If you're seeing the system dialog to enable notifications when the app launches in the emulator you must click one of the buttons manually otherwise the tests fail.
  3. ☑️ **Validate** the tests get past the landing screen successfully
- 2️⃣ Optionally to validate the e2e tests support the new landing screen in WordPress app, you can:
  - ✅ **Verify** the instrumented tests passed CI on PR: https://github.com/wordpress-mobile/WordPress-Android/pull/17871

## Regression Notes
1. Potential unintended areas of impact
   N/a

6. What I did to test those areas of impact (or what existing automated tests I relied on)
   Ran e2e and screenshots UI tests on:
    - Jetpack app
    - Wordpress app
       - with `LANDING_SCREEN_REVAMP` build config field `false`
       - with `LANDING_SCREEN_REVAMP` build config field `true`

7. What automated tests I added (or what prevented me from doing so)
   N/a

PR submission checklist:
